mupdf fails to cross-compile

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix build .#pkgsCross.aarch64-multiplatform.mupdf`

### Build log
```
       > source/fitz/draw-path.c:1051:8: note: 'old_bx' was declared here
       >  1051 |  float old_bx, old_by;
       >       |        ^~~~~~
       > source/fitz/harfbuzz.c:30:10: fatal error: hb.h: No such file or directory
       >    30 | #include "hb.h"
       >       |          ^~~~~~
       > compilation terminated.
       > make[1]: *** [Makefile:126: build/shared-release/source/fitz/harfbuzz.o] Error 1
       > make[1]: Leaving directory '/build/mupdf-1.20.3-source'
       > make: *** [Makefile:487: shared-release] Error 2
```

### Additional context
Mupdf executes `$(shell pkg-config ...`

--- a/pkgs/applications/misc/mupdf/default.nix
+++ b/pkgs/applications/misc/mupdf/default.nix
@@ -5,6 +5,7 @@
 , copyDesktopItems
 , makeDesktopItem
 , desktopToDarwinBundle
+, buildPackages
 , pkg-config
 , freetype
 , harfbuzz
@@ -49,13 +50,17 @@ stdenv.mkDerivation rec {
 
   postPatch = ''
     sed -i "s/__OPENJPEG__VERSION__/${openJpegVersion}/" source/fitz/load-jpx.c
+    substituteInPlace Makerules --replace "(shell pkg-config" "(shell $PKG_CONFIG"
   '';
 
   # Use shared libraries to decrease size
   buildFlags = [ "shared" ];
 
-  makeFlags = [ "prefix=$(out)" "USE_SYSTEM_LIBS=yes" ]
-    ++ lib.optionals (!enableX11) [ "HAVE_X11=no" ]
+  makeFlags = [
+    "prefix=$(out)"
+    "USE_SYSTEM_LIBS=yes"
+    "PKG_CONFIG=${buildPackages.pkg-config}/bin/${buildPackages.pkg-config.targetPrefix}pkg-config"
+  ] ++ lib.optionals (!enableX11) [ "HAVE_X11=no" ]
     ++ lib.optionals (!enableGL) [ "HAVE_GLUT=no" ];
 
   nativeBuildInputs = [ pkg-config ]

